app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java

```diff
@@ -0,0 +1,108 @@
+import com.google.droidjump.leaderboards_data.LeaderboardsScoresAdapter;
+import com.google.droidjump.models.NavigationHelper;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Displays Leaderboards Screen.
+ */
+public class LeaderboardScoresFragment extends Fragment {
+    private Leaderboard leaderboard;
+    private MainActivity activity;
+    private List<LeaderboardScore> scores;
+    private LeaderboardsScoresAdapter adapter;
+    private int timeSpan;
+    private int collection;
+
+    public LeaderboardScoresFragment(Leaderboard leaderboard) {
+        this.leaderboard = leaderboard;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        activity = (MainActivity) getActivity();
+        scores = new ArrayList<>();
+        timeSpan = LeaderboardVariant.TIME_SPAN_ALL_TIME;
+        collection = LeaderboardVariant.COLLECTION_PUBLIC;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.leaderboard_scores_screen, container, /* attachToRoot= */ false);
+        ImageManager.create(activity).loadImage((ImageView) rootView.findViewById(R.id.leaderboard_avatar), leaderboard.getIconImageUri());
+        ((TextView) rootView.findViewById(R.id.leaderboards_title)).setText(leaderboard.getDisplayName());
+        adapter = new LeaderboardsScoresAdapter(scores, activity);
+        RecyclerView scoresView = rootView.findViewById(R.id.scores_recycler_view);
+        scoresView.addItemDecoration(new DividerItemDecoration(activity, LinearLayoutManager.VERTICAL));
+        scoresView.setAdapter(adapter);
+        fetchScores(timeSpan, collection);
+        Switch friendsSwitch = rootView.findViewById(R.id.friends_switch);
+        friendsSwitch.setOnClickListener(ignored -> {
+            if (friendsSwitch.isChecked()) {
+                collection = LeaderboardVariant.COLLECTION_FRIENDS;
+            } else {
+                collection = LeaderboardVariant.COLLECTION_PUBLIC;
+            }
+            Log.d(getClass().toString(), String.valueOf(collection));
+            fetchScores(timeSpan, collection);
+        });
+        rootView.findViewById(R.id.back_button).setOnClickListener(ignored -> activity.onBackPressed());
+        NavigationHelper.addOnBackPressedEventListener(activity);
+        return rootView;
+    }
+
+    private void fetchScores(int timeSpan, int collection) {
+        Games.getLeaderboardsClient(activity, activity.getSavedSignedInAccount())
+                .loadPlayerCenteredScores(
+                        leaderboard.getLeaderboardId(), timeSpan,
+                        collection, GameConstants.SCORES_PER_PAGE, /* forceReload= */ false)
+                .continueWithTask(task -> {
+                    LeaderboardScoreBuffer scoreBuffer = task.getResult().get().getScores();
```

Comment:
what if it is not successsful? a in the call fails due to network error?

Reply:
Discussed on GVC.